packages/store/src/secure-store/index.ts

```diff
@@ -4,14 +4,59 @@ import { isAppStudio, errorString } from '../utils';
 import { DummyStore } from './dummy-store';
 import { KeytarStore } from './keytar-store';
 import type { SecureStore } from './types';
+import { join, dirname } from 'path';
+import { homedir } from 'os';
+import { default as fs } from 'fs';
+declare function __non_webpack_require__(m: string): any;
+
+function getKeytarPaths(insiders: boolean): string[] {
+    const vscodeRootPath = insiders ? '.vscode-insiders' : '.vscode';
+    const vscodeExtensionsPath = join(homedir(), vscodeRootPath, 'extensions/');
+    const AppMFoldersVscode =
+        fs
+            .readdirSync(vscodeExtensionsPath)
+            .filter((fn) => fn.startsWith('sapse.sap-ux-application-modeler-extension')) ?? [];
+    return AppMFoldersVscode.map((AppMFolderVscode) => {
+        const extensionPath = join(vscodeExtensionsPath, AppMFolderVscode);
+        const keytarPackageJsonPath = join(extensionPath, 'node_modules/keytar/package.json');
+        if (fs.existsSync(keytarPackageJsonPath)) {
+            return dirname(keytarPackageJsonPath);
+        } else {
+            return '';
+        }
+    }).filter((dirname) => dirname !== '');
+}
 
 function getKeytar(log: Logger): typeof Keytar | undefined {
     try {
         return require('keytar');
     } catch (err) {
         log.warn(errorString(err));
+        // Try to load keytar from sap-ux-application-modeler-extension node_modules if available this helps in some
+        // cases such as windows machines with restricted access. From node modules such as @sap/generator-fiori or @sap/ux-ui5-tooling
+        // keytar is not installed or is removed from the fs by virus scanner.
+        try {
+            const AppMKeytarDirs = getKeytarPaths(false).concat(getKeytarPaths(true));
+            log.info('keytarDirectories: \n' + JSON.stringify(AppMKeytarDirs));
+            if (AppMKeytarDirs.length > 0) {
+                // try to load keytar from the first directory found
+                const keytarDir = AppMKeytarDirs[0];
+                log.info('Try to load keytar from :' + JSON.stringify(keytarDir));
+                // Support bundling
+                if (typeof __non_webpack_require__ === 'function') {
```

Comment:
Wont this always be true since the function is declared above

Reply:
This is runtime type rather than typescript. So no, it is only true in bundle.

Reply:
I dont follow sorry, its defined here https://github.com/SAP/open-ux-tools/pull/2230/files#diff-b235f66bb5ed958a1d510e9bff09af9bc65446b6fe34d31815fcc09c818dd7aeR11 at the top of this file.

Reply:
Thx